api/envoy/extensions/filters/network/redis_proxy/v3/redis_proxy.proto

```diff
@@ -321,6 +322,11 @@ message RedisProxy {
   // client. If an AUTH command is received when the password is not set, then an "ERR Client sent
   // AUTH, but no ACL is set" error will be returned.
   config.core.v3.DataSource downstream_auth_username = 7 [(udpa.annotations.sensitive) = true];
+
+  // The DNS cache configuration that the filter will attach to and that will be used to resolve
+  // hostnames that are returned with MOVED and ASK responses. If not provided, no DNS lookups
+  // will be performed (and thus the MOVED/ASK errors will be propagated to the user).
+  common.dynamic_forward_proxy.v3.DnsCacheConfig dns_cache_config = 10;
```

Comment:
I just realized this should actually be another `ConnPoolSettings`, where `enable_redirection` is since a) DNS lookups will be handled by the ConnPool and b) it's a no-op to enable it if `enable_redirection` is `off`.

I'll fix it.

cc: @mattklein123